src/main/java/com/microsoft/graph/core/requests/GraphClientFactory.java

```diff
@@ -75,13 +75,13 @@ public static OkHttpClient.Builder create(@Nonnull BaseBearerTokenAuthentication
      */
     @Nonnull
     public static OkHttpClient.Builder create(@Nonnull GraphClientOption graphClientOption, @Nonnull Interceptor... interceptors) {
-        final OkHttpClient.Builder builder = create(graphClientOption);
+        final OkHttpClient.Builder builder = KiotaClientFactory.create(interceptors);
         //Skip adding interceptor if that class of interceptor already exist.
         final List<String> appliedInterceptors = new ArrayList<>();
         for(Interceptor interceptor: builder.interceptors()) {
             appliedInterceptors.add(interceptor.getClass().toString());
         }
-        for (Interceptor interceptor:interceptors){
+        for (Interceptor interceptor:createDefaultGraphInterceptors(graphClientOption)){
```

Comment:
Thanks for the changes @raghucha. We might have a bigger problem here.

I think this could lead to conflicts since the order of handlers in the pipeline matters. e.g. by default all retries/redirects should include query parameter name decoding etc. A potential customization of query parameter decoding could mean it's added before the retry & redirect handlers..

I think we should avoid filling in missing middleware totally in case a developer chooses to pass their own interceptors. This seems to align with .NET and other languages as well.

Thoughts @baywet @andrueastman?

Reply:
correct, order matters, and this change might impact it. This is what was done in dotnet to address the same concern. We should probably align with that instead.
https://github.com/microsoft/kiota-http-dotnet/pull/246

Reply:
Thanks for added context Vincent. Would you like to implement this instead @raghucha?

Reply:
i can implement it however i have few design question . Below is a sample code from KiotaClientFactory. If anyone uses below method to create a Client they will never be able to Override the default interceptor functionality .And Question such as if they use `addInterceptor` Method will also never be able to override .  Is this acceptable? 

 ```
 @Nonnull public static OkHttpClient.Builder create(
            @Nonnull final BaseBearerTokenAuthenticationProvider authenticationProvider) {
        ArrayList<Interceptor> interceptors = new ArrayList<>(createDefaultInterceptorsAsList());
        interceptors.add(new AuthorizationHandler(authenticationProvider));
        return create(interceptors);
    }
```

Reply:
The createDefaultInterceptorsAsList was added two days ago. I'm not sure why we made it public @Ndiritu  ?
If that was an oversight, I'd be ok with making it private again.

Regardless of whether we keep it public, we should be able to add overloads all the way down the stack that accept options. And simply point the current methods that do not accept the options to call those new overloads.

Let us know if you have any additional comments or questions.

Reply:
@baywet, this was intentional.

I added [this overload](https://github.com/microsoft/kiota-java/blob/main/components/http/okHttp/src/main/java/com/microsoft/kiota/http/KiotaClientFactory.java#L61) that accepts `List<Interceptor>`. Current setup involves manipulating array size to add items etc.

In the process exposed `createDefaultInterceptorsAsList()` to potentially let a developer alter the list and pass it to `create()`. Happy with reversing `createDefaultInterceptorsAsList()` but I found value in using `create(List<Interceptor>)`

Reply:
Apologies, I should have made this clearer in the PR description.

Reply:
are we using it outside of kiota java? (i.e. here in core)

Reply:
Added a similar `create(List<Interceptor>)` overload in Graph core [ref](https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/main/src/main/java/com/microsoft/graph/core/requests/GraphClientFactory.java#L49) but not `createDefaultInterceptorsAsList()`. The latter is only in Kiota Java.